### PR TITLE
docs:file hygeiene

### PR DESCRIPTION
**Pull request by pterw**
## Summary

 - See commits.

## Checklist

- [x] `pytest -q` passes (update SESSION_CONTEXT Section 1 if test count changed)
- [x] `pre-commit run --all-files` passes (all hooks)
- [x] `python scripts/doc_state_sync.py --check` exits 0
- [x] PLAYBOOK Section 3 reflects current WP/task state
- [x] PLAYBOOK Section 4 has a dated log entry:
      - Batch WP: inside `DOCSYNC:CURRENT-BATCH-START/END` markers
      - Side task: after `DOCSYNC:CURRENT-BATCH-END` marker
- [x] Changes are within scope of the active batch definition
      (or deviation is logged in the Section 4 entry)

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -38,25 +38,26 @@ these failure modes.
 
 The external-memory layer consists of five tracked files and two archive
 directories. Each has a primary concern, and the design goal is that
-canonical facts live in exactly one place. In practice there is some
-intentional overlap: `HANDOFF_PROMPT.md` condenses key rules and
+canonical facts live in exactly one place.
```
[the remark this answers is not in the document]
Comment (pterw) [reply DEVELOPMENT.md]: FINDINGS is not kept up-to-date as per agents, this is a stale file to refer to for fixes to be implemented.

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -190,14 +190,16 @@ entrypoint (`cli.py`), and typed dataclass models (`models.py`); the root
...
+rotation, dedup, cross-validation, and CLI modes.
+
+That count has grown since the original Batch 14 refactor as later batches added edge-case coverage;
 run `pytest tests/test_docsync_*.py -q` to confirm the current number rather
 than trusting any figure quoted here, since it will drift as new tests are
 added.
 
-**SESSION_CONTEXT.md is optional in CI.** The file is committed to the
-repo and is normally present in GitHub Actions (with a standard
+**SESSION_CONTEXT.md is optional in CI.**
```
[the remark this answers is not in the document]
Comment (pterw) [reply DEVELOPMENT.md:208]: @copilot resolve this

[the remark this answers is not in the document]
Comment (pterw) [reply DEVELOPMENT.md:208]: @claude  Resolve this issue

File: README.md [hunk 3]
```diff
@@ -160,35 +160,35 @@ loading.js polls GET /progress?job_id=...
...
-* **Bounded Job Concurrency:** `MAX_ACTIVE_JOBS` (default 10, env-tunable) caps simultaneous background jobs via a `BoundedSemaphore` in `scrobblescope/worker.py`. Requests beyond the cap are rejected at the route before any job is created, and the concurrency slot is always released -- even on thread-start failure.
 * **CSRF Protection:** All mutating POST routes (`/results_loading`, `/results_complete`, `/unmatched_view`, `/reset_progress`) are protected via Flask-WTF `CSRFProtect`. Form submissions include a hidden `csrf_token` input; programmatic POSTs from `loading.js` read a `<meta name="csrf-token">` tag and inject the token into both form bodies and the `X-CSRFToken` header.
 * **Doc-State Sync Tooling:** A modular Python package (`scripts/docsync/`) keeps orchestration docs (PLAYBOOK, SESSION_CONTEXT, archive) consistent across agent handoffs. Deterministic rotation, SHA-256 dedup, and cross-validation replace manual copy-paste that drifted in earlier sessions. See [DEVELOPMENT.md](DEVELOPMENT.md) for rationale.
```
[the remark this answers is not in the document]
Comment (pterw) [reply README.md:160]: @copilot keep the `doc-state sync tooling` item in key implementation highlights, remedy batch 20 WP-1

File: DEVELOPMENT.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (pterw) [reply DEVELOPMENT.md:208]: @copilot  Check if this issue has been resolved.

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # ScrobbleScope: Development Methodology
 
-This document explains how ScrobbleScope was built -- the orchestration
+This document explains how ScrobbleScope was built: the orchestration
```
[the remark this answers is not in the document]
Comment (pterw) [reply DEVELOPMENT.md:3]: @copilot resolve this comment

File: README.md [hunk 5]
```diff
@@ -235,14 +233,14 @@ loading.js polls GET /progress?job_id=...
     LASTFM_API_KEY="your_lastfm_api_key_here"
     SPOTIFY_CLIENT_ID="your_spotify_client_id_here"
     SPOTIFY_CLIENT_SECRET="your_spotify_client_secret_here"
+    SECRET_KEY="changeme_in_production"
```
[the remark this answers is not in the document]
Comment (pterw) [reply README.md]: @claude examine this comment and fix if relevant

**Review by pterw — COMMENTED** [review]
Please review comments, check if they are still valid.

File: DEVELOPMENT.md [hunk 6]
```diff
@@ -206,37 +216,65 @@ still occurs. See commit `05c7b19` on `main` for the original change.
 
 ---
 
+## Claude Code Skills (tightly scoped tooling)
```
[the remark this answers is not in the document]
Comment (pterw) [reply DEVELOPMENT.md:219]: @copilot Verify if this is resolved.

File: DEVELOPMENT.md [hunk 7]
```diff
@@ -206,37 +216,65 @@ still occurs. See commit `05c7b19` on `main` for the original change.
...
+
+Two project-scoped Claude Code (CC) skills provide structured entry points for
+common tasks. They are CC-specific; the portable, model-agnostic orchestration
+rules live in `AGENTS.md`.
```
[the remark this answers is not in the document]
Comment (pterw) [reply DEVELOPMENT.md]: @copilot Verify if this is important or resolved.

File: README.md [hunk 8]
```diff
@@ -117,41 +105,53 @@ This project was initially built to identify top albums released in a specific y
 
 ```mermaid
 graph LR
-    A[Browser] -->|form POST| B[routes.py]
+    A[Browser] -->|POST /results_loading| B[routes.py]
```
[the remark this answers is not in the document]
Comment (pterw) [reply README.md:93]: @copilot  Verify if this is important or resolved.

File: DEVELOPMENT.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (pterw) [reply DEVELOPMENT.md:3]: @copilot Verify the status of this, if it is resolved or not.

File: DEVELOPMENT.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (pterw) [reply DEVELOPMENT.md:219]: @copilot Verify if this is resolved.

File: scripts/docsync/renderer.py [hunk 9]
```diff
@@ -80,7 +80,7 @@ def _build_status_block(
             next_wp = f"WP-{candidate}"
         else:
             next_wp = "unknown"
-        newest_heading = current_entries[0].heading.removeprefix("### ").strip()
+        newest_heading = current_entries[-1].heading.removeprefix("### ").strip()
```
[the remark this answers is not in the document]
Comment (pterw) [reply scripts/docsync/renderer.py:83]: @claude REad AGENTS.MD then check if this is relevant, resolved, or needs attention.

File: .claude/SESSION_CONTEXT.md [hunk 10]
```diff
@@ -19,7 +19,7 @@ Last updated: 2026-07-23
 | Batch 17 status | **Complete**. All 4 WPs done (WP-5 dropped). Definition: `docs/history/definitions/BATCH17_DEFINITION.md`. |
 | Batch 18 status | **Complete**. All 5 WPs done. Definition: `docs/history/definitions/BATCH18_DEFINITION.md`. |
 | Batch 19 status | **Complete**. All 5 WPs done plus owner-review follow-up. Definition: `docs/history/definitions/BATCH19_DEFINITION.md`. PR #152 merged to `main`. |
-| Batch 20 status | **Active.** File-hygiene + docs methodology refresh. WP-0 done. Definition: `BATCH20_DEFINITION.md`. |
+| Batch 20 status | **Active.** File-hygiene + docs methodology refresh. WP-0 through WP-4 done. Definition: `BATCH20_DEFINITION.md`. |
```
[the remark this answers is not in the document]
Comment (pterw) [reply .claude/SESSION_CONTEXT.md]: @copilot Check if this is resolved or needs attention.

**Review by pterw — COMMENTED** [review]
Please review these comments and their importance, if they are fixed, or if they need attention.

File: DEVELOPMENT.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (pterw) [reply DEVELOPMENT.md]: @copilot Check if this has been resolved or if this is important or relevant.

[the remark this answers is not in the document]
Comment (pterw) [reply DEVELOPMENT.md]: @copilot Should this be ignored? Or should the text clarify it? Use your best judgment.

File: tests/test_docsync_renderer.py [hunk 11]
```diff
@@ -189,6 +189,32 @@ def test_entry_without_count_shows_unknown_count(self):
         text = "\n".join(block)
         assert "unknown" in text.lower()
 
+    def test_newest_heading_uses_last_entry(self):
```
[the remark this answers is not in the document]
Comment (pterw) [reply tests/test_docsync_renderer.py:192]: @copilot check if this is resolved and if not implemenmt the fix leave a reply, and read AGENTS.MD before starting

**Review by pterw — COMMENTED** [review]
Make sure this will be legible to agents in IDEs such as Claude Code and Codex, in VSCode.

File: AGENTS.md [hunk 12]
```diff
@@ -108,8 +147,9 @@ Conventional Commits, imperative mood, no trailing period:
 1. `pytest -q` -- all tests pass.
 2. `pre-commit run --all-files` -- all hooks pass.
 3. Stage only files changed for this work package.
-4. Commit after each WP (do not batch multiple WPs into one commit). Do not
-   push unless the owner explicitly asks for a push.
+4. Commit after each WP (do not batch multiple WPs into one commit). In
+   GitHub Copilot sessions, use the platform progress/reporting tool for
+   commit + push handoff; do not push directly with shell `git`/`gh`.
```
[the remark this answers is not in the document]
Comment (pterw) [reply AGENTS.md]: @claude[agent]  Read AGENTS.md and give a reply on if this should be comitted as a suggestion or not.

File: PLAYBOOK.md [hunk 13]
```diff
@@ -138,6 +138,65 @@ non-current operational logs. Older dated entries live in
 - Forward guidance: WP-1 starts the README pass (test-file count, project
   structure, mermaid diagram refresh).
 
+### 2026-07-24 - README counts/structure/mermaid refresh (Batch 20 WP-1)
```
[the remark this answers is not in the document]
Comment (pterw) [reply PLAYBOOK.md:141]: @copilot  Verify if this is valid, read AGENTS.MD. If so, fix it and leave a reply confirming. If not, leave a reply saying why it isn't valid.

File: PLAYBOOK.md [hunk 14]
```diff
@@ -138,8 +138,133 @@ non-current operational logs. Older dated entries live in
...
+- Deviations: none.
+- Validation: `pytest -q` -- **389 passed**. `pre-commit run --all-files` --
+  all hooks pass.
+- Forward guidance: WP-5 adds the Claude Code skills subsection in
+  `DEVELOPMENT.md`.
+
+### 2026-07-24 - DEVELOPMENT.md skills subsection (Batch 20 WP-5)
+
+- Scope: completed Batch 20 WP-5 -- added the "Claude Code Skills (tightly
+  scoped tooling)" subsection to `DEVELOPMENT.md`.
```
[the remark this answers is not in the document]
Comment (pterw) [reply PLAYBOOK.md:205]: @copilot Read AGENTS.MD and check if this is valid or not, if valid please rectify and leave a reply, and if not valid please leave a reply explaining why.

**Review by pterw — COMMENTED** [review]
Please verify these and make sure they are valid per AGENTS.md

File: README.md [hunk 15]
```diff
@@ -117,41 +117,53 @@ This project was initially built to identify top albums released in a specific y
...
+    A -->|POST /heatmap_loading| B
```
[the remark this answers is not in the document]
Comment (pterw) [reply README.md:94]: @claude Check if this is valid and worth considering, read AGENTS.MD. If it is valid please fix it and leave a reply. If it is not please reply explaining why.

**Review by pterw — COMMENTED** [review]
@copilot Check for duplications.

File: README.md [hunk 16]
```diff
@@ -279,16 +287,35 @@ python init_db.py
...
-* Docker installed and running.
-* The `ss-postgres` container must exist (created once via `docker run`; see
-  `AGENT_NOTES.md` Local Dev Setup section for the full command).
-* `DATABASE_URL` set in `.env`:
+* Docker must be installed and running.
+* The `ss-postgres` container must exist. Create it once with:
+  ```bash
+  docker run -d --name ss-postgres \
+      -e POSTGRES_PASSWORD=postgres -e POSTGRES_USER=postgres \
+      -e POSTGRES_DB=scrobblescope \
```
[the remark this answers is not in the document]
Comment (pterw) [reply README.md:265]: @claude Read AGENTS.MD and verify the validity of this. If it is to be fixed, do so and leafve a a reply. If it is not, reply why it is not worth it.

**Review by pterw — COMMENTED** [review]
Screenshots have been remove due to bloat.

**Review by pterw — COMMENTED** [review]
@copilot Check that everything is logical and improved readability, space, and hygiene. Run a pytest suite. @claude

File: README.md [hunk 17]
```diff
@@ -160,35 +145,35 @@ loading.js polls GET /progress?job_id=...
...
-* **Bounded Job Concurrency:** `MAX_ACTIVE_JOBS` (default 10, env-tunable) caps simultaneous background jobs via a `BoundedSemaphore` in `scrobblescope/worker.py`. Requests beyond the cap are rejected at the route before any job is created, and the concurrency slot is always released -- even on thread-start failure.
-* **CSRF Protection:** All mutating POST routes (`/results_loading`, `/results_complete`, `/unmatched_view`, `/reset_progress`) are protected via Flask-WTF `CSRFProtect`. Form submissions include a hidden `csrf_token` input; programmatic POSTs from `loading.js` read a `<meta name="csrf-token">` tag and inject the token into both form bodies and the `X-CSRFToken` header.
+* **CSRF Protection:** All mutating POST routes (`/results_loading`, `/heatmap_loading`, `/results_complete`, `/unmatched_view`, `/reset_progress`) are protected via Flask-WTF `CSRFProtect`. Form submissions include a hidden `csrf_token` input; programmatic POSTs from `loading.js` read a `<meta name="csrf-token">` tag and inject the token into both form bodies and the `X-CSRFToken` header.
```
[the remark this answers is not in the document]
Comment (pterw) [reply README.md]: @copilot Read AGENTS.MD and verify the veracity and importanceo f this finding. If it is valid, commit the suggestion or I will, and write a reply saying it is valid. If it is not, write a reply explaining it is not and why.

File: README.md [hunk 18]
```diff
@@ -160,35 +145,32 @@ loading.js polls GET /progress?job_id=...
...
-* **CSRF Protection:** All mutating POST routes (`/results_loading`, `/results_complete`, `/unmatched_view`, `/reset_progress`) are protected via Flask-WTF `CSRFProtect`. Form submissions include a hidden `csrf_token` input; programmatic POSTs from `loading.js` read a `<meta name="csrf-token">` tag and inject the token into both form bodies and the `X-CSRFToken` header.
+* **CSRF Protection:** All mutating POST routes (`/results_loading`, `/heatmap_loading`, `/results_complete`, `/unmatched_view`, `/reset_progress`) are protected via Flask-WTF `CSRFProtect`. Two complementary mechanisms are used: form-submit routes (`/results_complete`, `/results_loading`) include a hidden `csrf_token` body input; fetch-based routes read a `<meta name="csrf-token">` tag -- `/reset_progress` sends the token in the `X-CSRFToken` header only, while `/heatmap_loading` sends it in both the body and the header.
```
[the remark this answers is not in the document]
Comment (pterw) [reply README.md]: @copilot  Please read AGENTS.MD and give an a verdict on this suggestion.

**Comment by pterw** [comment]
@copilot please check if python scripts/doc_state_sync.py --check exits as expected.

**Comment by pterw** [comment]
@copilot finally please make sure to read AGENTS.MD and make sure the playbook and session_context reflect the state.

File: DEVELOPMENT.md [hunk 19]
```diff
@@ -206,37 +216,67 @@ still occurs. See commit `05c7b19` on `main` for the original change.
...
+common tasks. They are CC-specific; the portable, model-agnostic orchestration
+rules live in `AGENTS.md`. The skill definitions themselves are maintained
+locally and are not tracked in this repository (`.gitignore` excludes `.claude/`
+except `SESSION_CONTEXT.md`); this section documents their purpose for context.
+
+**`scrobblescope-bootstrap`** runs the canonical session bootstrap in a fixed
+read order: `.claude/SESSION_CONTEXT.md` Sections 1-2, then `PLAYBOOK.md`
+Sections 3-4, then `AGENT_NOTES.md`, stopping early if those three files
+confirm the current batch and next work package. Invoke it at the start of any
+substantive session -- new feature work, refactors, or multi-WP batch work.
```
[the remark this answers is not in the document]
Comment (pterw) [reply DEVELOPMENT.md:232]: @copilot Please evaluate the merit of this, and if it will increasse amnesiac agents or reduce token bloat. Be decisive and pick an option to commit or not.
